src/vmm/src/snapshot/mod.rs

```diff
@@ -177,55 +162,49 @@ impl Snapshot {
         }
 
         let mut snapshot_slice: &[u8] = snapshot.as_mut_slice();
-        Snapshot::unchecked_load::<_, O>(&mut snapshot_slice)
+        Snapshot::load_unchecked::<_>(&mut snapshot_slice)
     }
+}
 
-    /// Load a snapshot from a reader object and perform a snapshot version check
-    pub fn load_with_version_check<T, O>(
-        &self,
-        reader: &mut T,
-        snapshot_len: usize,
-    ) -> Result<O, SnapshotError>
+impl<Data: Serialize + Debug> Snapshot<Data> {
+    /// Helper function to serialize an object to a writer
+    pub fn serialize<T, O>(writer: &mut T, data: &O) -> Result<usize, SnapshotError>
```

Comment:
just like `deserialize`, this can be a free standing function, non-`pub`, and be reused in `SnapshotHdr`